src/Select.js

```diff
@@ -879,9 +880,17 @@ const Select = React.createClass({
 		}
 
 		for (var i = 0; i < options.length; i++) {
-			if (!options[i].disabled) return i;
+			// if there is no selected option just use first value in options
+			if (!options[i].disabled && !focusedOption) {
+				return i;
+			} else if (!options[i].disabled && options[i].value === focusedOption.value) {
```

Comment:
Instead of using the hard coded 'value' prop. Use the valueKey prop. That way it works when a non-standard valueKey is used.
`} else if (!options[i].disabled && options[i][this.props.valueKey] === focusedOption[this.props.valueKey]) {`